studio/components/interfaces/Settings/Logs/PreviewFilterPanel.tsx

```diff
@@ -97,9 +110,17 @@ const PreviewFilterPanel: FC<Props> = ({
       Refresh
     </Button>
   )
-  const handleSearch = (partial: Partial<Parameters<LogSearchCallback>[0]>) =>
-    onSearch({ query: search, to: partial?.to || null, from: partial?.from || null })
+  const handleSearch = (partial: Partial<Parameters<LogSearchCallback>[0]>) => {
+    const shouldShowUpgradePrompt = maybeShowUpgradePrompt(partial.from)
```

Comment:
it would be more appropriate to handle this restriction in triggering the query call in the parent component `<LogsPreviewer/ >`

Note that there is some refactoring of the querying logic in #8531